pkg/core/plugins/interfaces.go

```diff
@@ -80,3 +83,10 @@ type AuthnAPIServerPlugin interface {
 	Plugin
 	NewAuthenticator(PluginContext) (authn.Authenticator, error)
 }
+
+// PolicyPlugin a plugin to add a Policy to Kuma
+type PolicyPlugin interface {
+	Plugin
+	MatchedPolicies(dataplane *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error)
```

Comment:
not sure if `resources` here will be useful. Will policies written in this PolicyPlugin fashion be available in Resources in MeshContext?

Reply:
I believe they are as they end up being registered as part of the Schema like the others: https://github.com/kumahq/kuma/blob/5d376a6fe714255934f6371bea533c286d26d0ee/pkg/plugins/policies/scheme.go#L9-L16

Reply:
Ok